workflow/executor/agent.go

```diff
@@ -41,6 +42,7 @@ type AgentExecutor struct {
 	RESTClient        rest.Interface
 	Namespace         string
 	consideredTasks   map[string]bool
+	lock              sync.RWMutex
```

Comment:
there is a better approach, change `map[string]bool` to `sync.Map`, this does not risk programming errors

Reply:
https://pkg.go.dev/sync#Map

Reply:
ok